aiplatform/snippets/delete_dataset_test.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	aiplatform "cloud.google.com/go/aiplatform/apiv1"
+	aiplatformpb "cloud.google.com/go/aiplatform/apiv1/aiplatformpb"
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	"google.golang.org/api/option"
+)
+
+var (
+	projectID   string
+	location    string = "us-central1"
+	datasetName string
+	datasetID   string
+)
+
+func setup(t *testing.T) {
+	t.Helper()
+
+	// Create a new dataset
+	ctx := context.Background()
+	tc := testutil.SystemTest(t)
+
+	projectID = tc.ProjectID
```

Comment:
If there are multiple tests in parallel, they'll fight over the package-level `datasetName` and `datasetID` variables if `setup()` is called more than once.

can we make setup() return those values instead?

Reply:
if a test suite calls `t.Parallel()`, with multiple tests, it could happen. But now that's *2* things that have to go different for this to cause an issue. its probably safe enough as-is.